rest/route/host.go

```diff
@@ -145,50 +149,33 @@ func (h *hostChangeStatusHandler) Parse(ctx context.Context, r *http.Request) er
 		return errors.Wrap(err, "Argument read error")
 	}
 
-	if !util.StringSliceContains(evergreen.ValidUserSetStatus, h.Status) {
-		return fmt.Errorf("Invalid host status '%s' for host '%s'", h.Status, h.hostID)
-	}
-
 	return nil
 }
 
-func (h *hostChangeStatusHandler) Run(ctx context.Context) gimlet.Responder {
+func (h *hostModifyHandler) Run(ctx context.Context) gimlet.Responder {
 	user := MustHaveUser(ctx)
+	env := evergreen.GetEnvironment()
+	queue := env.RemoteQueue()
+
+	// Find host to be modified
 	foundHost, err := h.sc.FindHostByIdWithOwner(h.hostID, user)
 	if err != nil {
 		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error for find() by distro id '%s'", h.hostID))
 	}
 
-	if foundHost.Status == evergreen.HostTerminated {
-		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
-			StatusCode: http.StatusBadRequest,
-			Message:    fmt.Sprintf("Host '%s' is terminated; its status cannot be changed", foundHost.Id),
-		})
-	}
-
-	if h.Status == evergreen.HostTerminated {
-		if err = h.sc.TerminateHost(ctx, foundHost, user.Id); err != nil {
-			return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
-				StatusCode: http.StatusInternalServerError,
-				Message:    err.Error(),
-			})
-		}
-
-	} else {
-		if err = h.sc.SetHostStatus(foundHost, h.Status, user.Id); err != nil {
-			return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
-				StatusCode: http.StatusInternalServerError,
-				Message:    err.Error(),
-			})
-		}
+	changes := host.HostModifyOptions{
+		AddInstanceTags:    h.AddInstanceTags,
+		DeleteInstanceTags: h.DeleteInstanceTags,
 	}
 
-	host := &model.APIHost{}
-	if err = host.BuildFromService(foundHost); err != nil {
-		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "API Error converting from host.Host to model.APIHost"))
+	// Create new spawnhost modify job
```

Comment:
Before modifying the host shouldn't we check UserHost to verify this was spawned by a user? And perhaps check the user if possible.

Reply:
Potentially helpful: https://github.com/evergreen-ci/evergreen/blob/c3ede857bc7689be715622fb613800efc4cd713a/service/spawn.go#L187

Reply:
I think FindHostByIdWithOwner handles this upon receiving the patch request.